src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -115,8 +118,34 @@ private bool CouldBeAwaited(BoundExpression expression)
                 return false;
             }
 
+            if (containingMethod is SynthesizedSimpleProgramEntryPointSymbol { IsAsync: false })
+            {
+                var bag = BindingDiagnosticBag.GetInstance(withDependencies: false, withDiagnostics: true);
+                try
+                {
+                    _ = BindAwait(expression, expression.Syntax, bag);
+                    // Top-level statements are always allowed to use `await`, so we
+                    // can ignore specific errors
+                    Debug.Assert(bag.DiagnosticBag is not null);
+                    return !bag.DiagnosticBag.AsEnumerableWithoutResolution().Any(isErrorOtherThanBadAwait);
```

Comment:
>!bag.DiagnosticBag.AsEnumerableWithoutResolution().Any(isErrorOtherThanBadAwait); [](http://example.com/codeflow?start=27&length=81)

Perhaps the logic could be easier to follow if we remove double negative (remove negation at the top, change Any to All, etc.). It looks like we we are saying that the expression CouldBeAwaited when all errors that we get here are either ```ErrorCode.ERR_BadAwaitWithoutAsync```, or ```ErrorCode.ERR_BadAwaitWithoutVoidAsyncMethod```.